packages/workflow/src/records/handler/create.ts

```diff
@@ -163,6 +179,31 @@ async function createRecord(
   return mergeBundles(record, practitionerResourcesBundle)
 }
 
+type CreatedRecord =
+  | InProgressRecord
+  | ReadyForReviewRecord
+  | ValidatedRecord
+  | WaitingForValidationRecord
+
+function getEventAction(record: CreatedRecord) {
+  if (isInProgress(record)) {
+    return 'sent-notification'
+  }
+  if (isReadyForReview(record)) {
+    return 'sent-notification-for-review'
+  }
+  if (isValidated(record)) {
+    return 'sent-for-approval'
+  }
+  if (isWaitingExternalValidation(record)) {
+    return 'waiting-external-validation'
+  }
+  // type assertion
+  record satisfies never
```

Comment:
Interesting syntax! Learned something new, but why is it needed here?

Reply:
This will make it so that the compiler would complain if there was any unhandled case